core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -60,6 +60,14 @@ class TransactionStateManager(brokerId: Int,
 
   this.logIdent = "[Transaction Log Manager " + brokerId + "]: "
 
+  type WriteTxnMarkers = (String, // transactionalId
+    Long, // pid
+    Short, // producerEpoch
+    TransactionState, // currentState
+    TransactionMetadata, // current metadata
+    Int) // coordinatorEpoch
+    => Unit
```

Comment:
This is a function from 6 parameters to Unit. There are so many parameters that we added a comment next to each to give it a name. Well, that's just what a case class gives you! So it could be a function from a case class with 6 parameters to Unit. Does that make sense?